### PR TITLE
Migrate to pyproject.toml

### PR DESCRIPTION
**Pull request by paddatrapper**
Closes: #249

This seems to work on my system?

The upstream documentation can be found at:

* https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html
* https://peps.python.org/pep-0621

### CODE REVIEW

**Review by alerque — APPROVED** [review]

**Comment by paddatrapper** [comment]
@alerque missed the GitHub Actions calls, should all be there now

**Review by NMarkgraf — APPROVED** [review]
Runs like a champ for me.

File: README.md [hunk 1]
```diff
@@ -74,8 +74,8 @@ Note: pandoc 2.10 is short lived and 2.11 has minor API changes comparing to tha
 
 After cloning the repo and opening the panflute folder, run
 
-- `python setup.py install` to install the package locally
-- `python setup.py develop` to install locally with a symlink so changes are automatically updated
+- `python install .` to install the package locally
```
Comment (acederberg) [inline README.md:77]: Did you mean ``python -m pip install .``? There is no subcommand ``install`` of the ``python`` command.

Comment (paddatrapper) [reply README.md]: Yes. Thanks for spotting, I have fixed both references

**Pull request merged**